python/common/org/python/types/List.java

```diff
@@ -167,8 +168,14 @@ public List(org.python.Object [] args, java.util.Map<java.lang.String, org.pytho
             int count = Math.min(size, otherSize);
 
             for (int i = 0; i < count; i++) {
-                org.python.types.Bool b =
-                    (org.python.types.Bool) this.value.get(i).__lt__(otherList.value.get(i));
+                org.python.Object result = this.value.get(i).__lt__(otherList.value.get(i));
```

Comment:
I see! Hmm, `__SortedObjectComparator.compare()` could do with some refactoring since the logic seems to be repeated twice inside after all, and I wouldn't want to instantiate what looks like a inner utility class elsewhere.

However, now that I'm looking at it, `__SortedObjectComparator.compare()` plus all the `List` dunder comparison methods _don't_ implement the subclass priority rule, so they might be faulty on top of not being DRY. If CPython _does_ do all that, I guess the `org.python.types.Object.__cmp__()` helper method might have to be revived...